app/grants/views.py

```diff
@@ -117,19 +120,23 @@ def grant_fund(request, grant_id):
     # make sure a user can only create one subscription per grant
     if request.method == 'POST':
         subscription = Subscription()
-        # subscriptionHash and ContributorSignature will be given from smartcontracts and web3
-        # subscription.subscriptionHash = request.POST.get('input_name')
-        # subscription.contributorSignature = request.POST.get('description')
-        # Address will come from web3 instance
-        # subscription.contributorAddress = request.POST.get('reference_url')
+
+        print("it fired")
+        print(request.POST)
+
+
+        subscription.subscription_hash = request.POST.get('subscription_hash')
+        subscription.contributor_signature = request.POST.get('signature')
+        subscription.contributor_address = request.POST.get('contributor_address')
         subscription.amount_per_period = request.POST.get('amount_per_period')
-        # subscription.tokenAddress = request.POST.get('denomination')
+        subscription.token_address = request.POST.get('token_address')
         subscription.gas_price = request.POST.get('gas_price')
-        # network will come from web3 instance
-        # subscription.network = request.POST.get('amount_goal')
+        subscription.network = request.POST.get('network')
         subscription.contributor_profile = profile
         subscription.grant = grant
         subscription.save()
+        return redirect(f'/grants/{grant.pk}')
```

Comment:
Can you use `reverse()` with the `redirect()` here, so if the route changes, the view doesn't need updated?